### ISSUE TITLE
spi_mcux_ecspi does not handle gpio chipselect correctly

### ISSUE THREAD

**Issue by voltamund** [zephyrproject-rtos/zephyr#83331]
**Describe the bug**

Target platform: MIMX8MM6 Cortex M4
If I use more than 4 gpios in the spi node's `cs-gpios` the driver it will not handle gpio with index > 3.
https://github.com/zephyrproject-rtos/zephyr/blob/6dce23326501a0850d2af9ae2af791ec0e7b6037/drivers/spi/spi_mcux_ecspi.c#L166
will fail. If the spi node uses `cs-gpios` it should just use `kECSPI_Channel0`. I am not even sure that using something like `kECSPI_Channel3` will work for the built-in cs.

**Expected behavior**
SPI transfer with `cs-gpios` index > 3 succeeds

**Logs and console output**
Slave 4 is greater than 3

**Environment (please complete the following information):**

 - OS: Linux
 - Toolchain: zephyr-sdk-0.16.8
 - Zephyr Version 3.7.0

**Comment by pdgendt** [comment]
You are correct, moreover this the case in many driver implementations. An index is needed and used to select the corresponding "hard-wired" chip-select OR the corresponding GPIO.
Attempting to support both cases in the same driver has led to this limitation, sadly.
